.github/workflows/tox_verify.yaml

```diff
@@ -1,6 +1,14 @@
 name: Tox
 
-on: [push, pull_request, workflow_dispatch]
+on:
+  push:
+    branches: [main]
+  pull_request:
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
```

Comment:
> Adding concurrency control to cancel in-progress runs when new commits are pushed

I think this is ok. Only your last commit gets checked if previous commits are interrupted. But might be too aggressive. Wdyt?

Reply:
`${{ github.head_ref || github.run_id }}` uses the PR branch name if it exists, otherwise falls back to the unique run ID. So, yes, for cron jobs. But also for manual triggers (workflow dispatch), perhaps other events too.